src/main/scala/com/monsanto/arch/cloudformation/model/simple/Builders.scala

```diff
@@ -244,6 +181,18 @@ trait Subnet extends AvailabilityZone with Outputs {
 
     f(sub) ++ sub.andOutput(name, name)
   }
+
+  def nat(routeTables: Seq[`AWS::EC2::RouteTable`])(implicit s: `AWS::EC2::Subnet`) = {
```

Comment:
I think the Subnet one makes sense as an implicit since this function is used nested within a subnet.  The VPC Gateway attachment isn't in that hierarchy though, so i'd make it explicit.